src/towncrier/_settings/load.py

```diff
@@ -146,14 +152,20 @@ def parse_toml(base_path: str, config: Mapping[str, Any]) -> Config:
 
     template = config.get("template", _template_fname)
     if template.startswith("towncrier:"):
-        resource_name = "templates/" + template.split("towncrier:", 1)[1] + ".rst"
-        if not pkg_resources.resource_exists("towncrier", resource_name):
+        file_manager = ExitStack()
+        atexit.register(file_manager.close)
+
+        resource_name = f"templates/{template.split('towncrier:', 1)[1]}.rst"
+        resource_path = resources.files("towncrier") / resource_name
+        file_manager.enter_context(resources.as_file(resource_path))
```

Comment:
you need something like resource_path = file_manager.enter_context(resources.as_file(resource_path))